### ISSUE TITLE
API Pages in PowerBI

### ISSUE THREAD

**Issue by nathannn-b** [hougaard/Simple-Object-Designer#117]
I have created a new API page and deployed the app in our production environment, but I cannot find the API page anywhere to be selected for use in PowerBI. 
I previously done the same thing in our Sandbox environment and this shows as expected under the hougaard API folder, but not for the production environment. 
I am wondering if I am missing any steps or need to do anything after creating the new API page and publishing the extension?

**Comment by hougaard** [comment]
No, nothing should be different, are you sure it's deployed to production? (check extension management and verify that your app is installed).

**Comment by nathannn-b** [comment]
Thanks for the quick reply! It is definitely deployed - checked extension management and also the deployment status via SOD. 
![image](https://user-images.githubusercontent.com/118318339/202049050-c8eef995-ae88-44a5-87dc-ed919bcb588d.png)

**Comment by nathannn-b** [comment]
![image](https://user-images.githubusercontent.com/118318339/202049135-2320258a-092d-476b-812d-32c27261680b.png)

**Comment by nathannn-b** [comment]
Here's what I see for our Sandbox Vs Production in PowerBI 
![image](https://user-images.githubusercontent.com/118318339/202049402-2bde8714-4b71-4cef-8a61-a8a6f5fdb0f9.png)
![image](https://user-images.githubusercontent.com/118318339/202049459-58825bde-46d9-4d97-af37-4c87cd872e59.png)

**Comment by hougaard** [comment]
I wonder if this could be a permissions issue?

How did you deploy to prod, from Simple Object Designer, or did you upload the .app file you got from your sandbox?

**Comment by nathannn-b** [comment]
I deployed directly from simple object designer in both environments. The methods I took to create the pages didn’t change across the environments and I am a super user in both environments also.

I have successfully used the sandbox apis in power automate flows also but I can’t find the api pages in power automate for the production environment either.

Sent from my iPhone
________________________________

**Comment by hougaard** [comment]
Weird, perhaps hit publish in SOD again in production .. This is not something I have ever heard from others before.

**Comment by nathannn-b** [comment]
I think I have found something that could be causing the issue. 
When I download the source code from the production environment, the launch.json file shows the environment type as sandbox:

            "name": "Microsoft cloud sandbox",
            "request": "launch",
            "type": "al",
            "environmentType": "Sandbox",

When I try to download symbols in vscode, I get the following error: 

[2022-11-16 10:53:22.46] Targeting Dynamics 365 Business Central environment tenant ID 'xxxxxxxx' and primary Azure AD tenant domain name ''.
[2022-11-16 10:53:22.47] Sending request to https://api.businesscentral.dynamics.com............
[2022-11-16 10:53:43.81] Error: An error occured while processing the request.
Request ID: xxxxxxxxxxxxxx
Session ID: xxxxxxxxxxxxxx

If you are targeting a cloud instance, supply these IDs if contacting Microsoft support.
[2022-11-16 10:53:43.82] The request for path /v2.0......... failed with code GatewayTimeout. Reason: Gateway Timeout
[2022-11-16 10:53:43.83] Sending request to https://api.businesscentral.dynamics.com/v2.0.........
[2022-11-16 10:54:26.23] Error: An error occured while processing the request.
Request ID: xxxxxxxxxxxxxxxxxx
Session ID: xxxxxxxxxxxxxxxxxx

Have you ever seen this or do you have any idea why the source code is showing a sandbox environment instead of production?

**Comment by hougaard** [comment]
That doesn't matter, from VS Code you can ONLY deploy to sandboxes.

**Comment by nathannn-b** [comment]
ok, so where do you propose I go from here? 
I have a project that I was counting on this extension to make happen, that was successful in Sandbox, but hasn't worked in production? How else can we troubleshoot this? Do you have any other suggestions for what I can check?

**Comment by hougaard** [comment]
Please try to uninstall the app from Extension Management and re-publish it from the Object Designer. This works for 100s of other users, so it should for you too..

**Comment by hougaard** [comment]
.. otherwise, we need to do a quick screenshare session...

**Comment by hougaard** [comment]
Did you find a solution?

**Comment by nathannn-b** [comment]
Hi - I completely uninstalled the SOD app, reinstalled it, created a new extension with a new API page, published it, and it worked. I then went and added more new API pages, republished my extension, and everything disappeared. I'm now back to square one with it

**Comment by hougaard** [comment]
Can we do a screenshare session later today?
